fhir-query-helper-base/src/main/java/io/elimu/a2d2/cds/fhir/helper/QueryingServerHelperBase.java

```diff
@@ -325,12 +330,27 @@ protected <S> FhirResponse<S> runWithInterceptors(QueryingCallback<S> callback,
 				client.unregisterInterceptor(correlationId);
 				client.unregisterInterceptor(singleAuth);
 				client.unregisterInterceptor(tracker);
+				if (noCache != null) {
```

Comment:
Should all of these `unregisterInterceptor` calls be in the `finally` block? What happens to them if an exception occurs?

Reply:
Will add local try / finally block, since the important thing is that the calls are done within the client.unwrap() synchronized block to avoid multi-thread inter-use.

The current implementation has this method being caled by the sub classes, each of which handles exceptions.